src/engine/plugins/tile.cpp

```diff
@@ -831,6 +831,53 @@ void encodeVectorTile(const datafacade::ContiguousInternalMemoryDataFacadeBase &
                 values_writer.add_float(util::vector_tile::VARIANT_TYPE_FLOAT, value);
             }
         }
+
+        // OSM Node tile layer
+        {
+            protozero::pbf_writer point_layer_writer(tile_writer, util::vector_tile::LAYER_TAG);
+            point_layer_writer.add_uint32(util::vector_tile::VERSION_TAG, 2);       // version
+            point_layer_writer.add_string(util::vector_tile::NAME_TAG, "osmnodes"); // name
+            point_layer_writer.add_uint32(util::vector_tile::EXTENT_TAG,
+                                          util::vector_tile::EXTENT); // extent
+
+            std::unordered_set<NodeID> internal_node_set;
+            for (const auto &edge : edges)
```

Comment:
Not really a problem, but I'd prefer one of a few other methods:

- std::vector + std::sort + std::unique

- std::set

Right now we get the slower unordered set for de-duplication, create a vector and sort after. Using a set directly would give you easy access to the sorted data. So you wouldn't go through multiple datasources here to get a sorted and unique set of entries.

The entire `internal_node_set` + `internal_nodes` process could be simplified to creating a set directly and loop over the set instead of the `internal_nodes` vector.

The `sort` + `unique` part would require twice as much memory for the initial vector, would would avoid creating the additional data structure.